aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp

```diff
@@ -271,7 +286,13 @@ namespace Aws
 
             void OpenSSLCipher::Init()
             {
-                EVP_CIPHER_CTX_init(&m_ctx);
+#if OPENSSL_VERSION_NUMBER < 0x10100003L
+                m_ctx = &_m_ctx;
+                EVP_CIPHER_CTX_init(m_ctx);
```

Comment:
Thanks, well spotted. EVP_CIPHER_CTX_init() is obsolete in OpenSSL 1.1, and they
#  define EVP_CIPHER_CTX_init(c)      EVP_CIPHER_CTX_reset(c)
in evp.h
So the call should really be to EVP_CIPHER_CTX_reset() instead. Can you apply that change before merging, or do you want me to update the commit?